src/Idris/Elab/Interface.idr

```diff
@@ -238,6 +232,20 @@ getMethToplevel {vars} env vis iname cname constraints allmeths params sig
     methName : Name -> Name
     methName n = UN (Basic $ bindName n)
 
+    collectImplicitNames : RawImp -> List Name
+    collectImplicitNames (IPi _ _ Explicit _        _ ty) = []
+    collectImplicitNames (IPi _ _ _        (Just n) _ ty) = n :: collectImplicitNames ty
+    collectImplicitNames (IPi _ _ _        Nothing  _ ty) = collectImplicitNames ty
+    collectImplicitNames _                                = []
+
+    namesToRawImp : Bool -> List Name -> RawImp -> RawImp
+    namesToRawImp bind (nm@(UN{}) :: xs) fn =
+      namesToRawImp bind xs $ INamedApp vfc fn nm $
+        if bind
+           then IBindVar vfc $ nameRoot nm
```

Comment:
`IBindVar` is needed so that using capitalized names does not generate an error. Such names occur in the `interface010` and `interface011` tests.

To avoid a name conflict, I'd like to use machine generated names, but `IBindVar` accepts a string.

Reply:
We can revisit this once #3526 is merged.